### PR TITLE
Add some benchmarks from the 'Are We Fast Yet' repo

### PR DESCRIPTION
**Pull request by fsfod**
Add some benchmarks from [Are We Fast Yet](https://github.com/smarr/are-we-fast-yet) repo that includes

- Havlak: A loop recognition algorithm.
- DeltaBlue: A constraint solver.
- Collisiondetector: A simulation of an airplane collision detector, based on WebKit's JavaScript CDjs.
- Bounce: Simulates some balls bouncing within a box that don't interact with each other.
- Queens: Solves the eight queens problem.

### CODE REVIEW

File: benchmarks/collisiondetector/bench.lua [hunk 1]
```diff
@@ -20,6 +20,10 @@
 -- OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 -- THE SOFTWARE.
 
+if pcall(require, "jit.opt") then
+  require("jit.opt").start("maxmcode=8000", "maxtrace=5000")
```
Comment (ltratt) [inline benchmarks/collisiondetector/bench.lua]: I think the rest of this file uses 4 space indents.

File: benchmarks/deltablue/bench.lua [hunk 2]
```diff
@@ -737,15 +737,14 @@ end
...
     Planner.chain_test(inner_iterations)
     Planner.projection_test(inner_iterations)
     return true
 end
 
-end -- object deltablue
-
-return deltablue
+function run_iter(n)
+  return (deltablue:inner_benchmark_loop(n))
```
Comment (ltratt) [inline benchmarks/deltablue/bench.lua]: 4 space indent?

File: benchmarks/queens/bench.lua [hunk 3]
```diff
@@ -35,14 +34,18 @@ function queens:verify_result (result)
...
     self.free_maxs  = {true, true, true, true, true, true, true, true,
                        true, true, true, true, true, true, true, true}
     self.free_mins  = {true, true, true, true, true, true, true, true,
                        true, true, true, true, true, true, true, true}
     self.queen_rows = {-1,   -1,   -1,   -1,   -1,   -1,   -1,   -1}
-    return self:place_queen(1)
+end
+
+function queens:setfirst(row)
+  self.queen_rows[row] = 1
```
Comment (ltratt) [inline benchmarks/queens/bench.lua]: 4 space indent?

File: benchmarks/bounce/bench.lua [hunk 4]
```diff
@@ -90,10 +88,8 @@ function bounce:benchmark ()
     return bounces
 end
 
-function bounce:verify_result (result)
-    return 1331 == result
+function run_iter(n)
+  local bounces = benchmark(n)
```
Comment (ltratt) [inline benchmarks/bounce/bench.lua]: 4 space indent?

**Comment by ltratt** [comment]
A few minor formatting consistency questions, but basically this is almost ready.

**Comment by fsfod** [comment]
@ltratt I fixed all the formatting you mentioned.

**Comment by ltratt** [comment]
OK, let's squash that down (you'll probably need to break that last commit up into pieces) and then I can merge.

**Comment by fsfod** [comment]
I was just going squash it down as two commits. The unmodified benchmarks and then one with them all changed.

**Comment by ltratt** [comment]
Works for me.

**Comment by fsfod** [comment]
@ltratt Squashed and ready to merge.

**Pull request merged**